src/match_traildb.c

```diff
@@ -1136,6 +1152,11 @@ int parse_args(int argc, char **argv,
           case 'f': *filter = optarg; break;
           case 'w': *window_file = optarg; break;
           case 'o': *format = optarg; break;
+          case '?':
+          case 'h':
+            print_usage(stderr, argv[0]);
+            exit(EX_USAGE);
```

Comment:
True. The `case '?'` clause for unhandled arguments will fall through to the `case 'h'`.
My thought is that invocations that result in the usage printed indicate bad arguments and explicitly asking for the usage is a special case.